lua/EnemyHealthbar.lua

```diff
@@ -327,10 +327,14 @@ elseif string.lower(RequiredScript) == "lib/units/beings/player/states/playersta
 				end
 			end
 
-			if not visible and self._last_unit and alive( self._last_unit ) then
+			if not visible and self._last_unit and alive( self._last_unit ) and self._last_unit:character_damage() then
 				health = self._last_unit:character_damage()._health * 10 or 0
-				max_health = self._last_unit:character_damage()._HEALTH_INIT * 10 or 0
-				name_id = self._last_unit:base()._tweak_table or "ENEMY"
+				max_health = (self._last_unit:character_damage()._HEALTH_INIT or 0) * 10
+				if not self._last_unit:base() then 
+					name_id = "ENEMY"
+				else
+					name_id = self._last_unit:base()._tweak_table or "ENEMY"
+				end
```

Comment:
this whole if-then-else-block can be simplified as a one-liner as well:
`name_id = self._last_unit:base() and self._last_unit:base()._tweak_table or "ENEMY"`

Reply:
Nah, sorry, I'm not that good at lua, haven't use it for ages =)